catalog/plugin/catalog-plugin-metacard-validation/src/main/java/ddf/catalog/metacard/validation/MetacardValidityMarkerPlugin.java

```diff
@@ -39,92 +45,89 @@
-                        boolean validationErrorsExist = e.getErrors() != null && !e.getErrors()
-                                .isEmpty();
-                        boolean validationWarningsExist =
-                                e.getWarnings() != null && !e.getWarnings()
-                                        .isEmpty();
-                        if (validationErrorsExist || validationWarningsExist) {
-                            // Check for warnings and errors
-                            if (validationErrorsExist) {
-                                validationErrors.addAll(e.getErrors());
-                            }
-                            if (validationWarningsExist) {
-                                validationWarnings.addAll(e.getWarnings());
-                            }
-                        } else {
-                            LOGGER.error(
-                                    "Metacard validator {} did not have any warnings or errors but it threw a validation exception."
-                                            + " There is likely something wrong with your implementation. This will result in the metacard not"
-                                            + " being properly marked as invalid.",
-                                    getValidatorName(metacardValidator));
-                        }
-
-                    } else {
-                        // If validator is explicitly turned on, break out and do not include in the
-                        // list of metacards that will be allowed through.
-                        toReturn = false;
-                        break;
-                    }
+    @Override
+    public UpdateRequest process(UpdateRequest input)
+            throws PluginExecutionException, StopProcessingException {
+        List<Map.Entry<Serializable, Metacard>> validatedMetacards =
+                getValidatedMetacards(input.getUpdates(), Map.Entry::getValue);
+        return new UpdateRequestImpl(validatedMetacards,
+                input.getAttributeName(),
+                input.getProperties(),
+                input.getStoreIds());
+    }
 
+    private <T> List<T> getValidatedMetacards(List<T> requestItems,
+            Function<T, Metacard> itemToMetacard) {
+        return requestItems.stream()
+                .map(item -> validate(item, itemToMetacard))
+                .filter(didNotFailEnforcedValidator)
+                .collect(Collectors.toList());
+    }
+
+    private <T> T validate(T item, Function<T, Metacard> itemToMetacard) {
+        Set<String> errors = new HashSet<>();
+        Set<String> warnings = new HashSet<>();
+
+        Metacard metacard = itemToMetacard.apply(item);
+        for (MetacardValidator validator : metacardValidators) {
+            try {
+                validator.validate(metacard);
+            } catch (ValidationException e) {
+                if (isValidatorEnforced(validator)) {
+                    return null;
```

Comment:
That's a good point; I actually ran into this issue a few days ago, but since I was familiar with how this all worked it wasn't too confusing. I can absolutely see this being a total blocker for someone who isn't as familiar though. A little line in the ingest log just saying "x metacard failed because the y validator is enforced" would be a great addition
Taking it even further, if we could say in the ingest log how many metacards were marked with errors, warnings, or both would be great - if you don't get all the results you expect after ingesting, it can look like something is wrong, but if we say in the ingest log "hey everything is there it's just invalid" it would probably help a lot